### PR TITLE
fix arp_neighbor_noptf for multi asic devices

### LINKED ISSUE #3671: test_neighbor_mac_noptf failing in multi asic devices
**Description**

The test in file `arp/test_neighbor_mac_noptf` are failing on multi asic platforms. The tests fail with following error
```
self = <tests.arp.test_neighbor_mac_noptf.TestNeighborMacNoPtf instance at 0x7f11261d6780>
duthosts = <tests.common.devices.duthosts.DutHosts object at 0x7f1125516250>
enum_rand_one_per_hwsku_frontend_hostname = 'svcstr-nsonic-acs-1'

    @pytest.fixture(scope="module", autouse=True)
    def setupDutConfig(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
        """
            Disabled BGP to reduce load on switch and restores DUT configuration after test completes
    
            Args:
                duthost (AnsibleHost): Device Under Test (DUT)
    
            Returns:
                None
        """
        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
        duthost.command("sudo config bgp shutdown all")
        if not wait_until(120, 2.0, self._check_no_bgp_routes, duthost):
>           pytest.fail('BGP Shutdown Timeout: BGP route removal exceeded 120 seconds.')
E           Failed: BGP Shutdown Timeout: BGP route removal exceeded 120 seconds.

duthost    = <MultiAsicSonicHost> svcstr-nsonic-acs-1
duthosts   = <tests.common.devices.duthosts.DutHosts object at 0x7f1125516250>
enum_rand_one_per_hwsku_frontend_hostname = 'svcstr-nsonic-acs-1'
self       = <tests.arp.test_neighbor_mac_noptf.TestNeighborMacNoPtf instance at 0x7f11261d6780>

```
The PR https://github.com/Azure/sonic-mgmt/pull/3369 introduced the change to count the routes. This change doesn't work on multi asic platform.

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshminarasimhan <arlakshm@microsoft.com>

### Description of PR

Summary:
Closes #3671

### Type of change

- [ x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [x ] 201911

#### What is the motivation for this PR?
This change fixes the issue https://github.com/Azure/sonic-mgmt/issues/3671

#### How did you do it?
Add support to the count_routes for all the asics

### CODE REVIEW

File: tests/arp/test_neighbor_mac_noptf.py [hunk 1]
```diff
@@ -30,22 +30,33 @@ class TestNeighborMacNoPtf:
         6: {"intfIp": "fe00::1/64", "NeighborIp": "fe00::2"},
     }	
 
-    def count_routes(self, host, prefix):
+    def count_routes(self, asichost, prefix):
         # Counts routes in ASIC_DB with a given prefix
-        num = host.shell(
-                'sonic-db-cli ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(ROUTE_TABLE_NAME, prefix),
+        num = asichost.shell(
+                '{} ASIC_DB eval "return #redis.call(\'keys\', \'{}:{{\\"dest\\":\\"{}*\')" 0'.format(asichost.sonic_db_cli,ROUTE_TABLE_NAME, prefix),
```
Comment (smaheshm) [inline tests/arp/test_neighbor_mac_noptf.py:36]: nit: space after ","

File: tests/arp/test_neighbor_mac_noptf.py [hunk 2]
```diff
@@ -30,22 +30,33 @@ class TestNeighborMacNoPtf:
...
                 module_ignore_errors=True, verbose=True)['stdout']
         return int(num)
 
-    def _check_no_bgp_routes(self, duthost):
-        # Checks that there are no routes installed by BGP in ASIC_DB by filtering out all local routes installed on testbeds
-        localv6 = self.count_routes(duthost, "fc") + self.count_routes(duthost, "fe")
-        localv4 = self.count_routes(duthost, "10.") + self.count_routes(duthost, "192.168.0.")
-        allroutes = self.count_routes(duthost, "")
-        bgproutes = allroutes - localv6 - localv4 - DEFAULT_ROUTE_NUM
+    def _check_no_bgp_routes_asic(self, asichost):
```
Comment (smaheshm) [inline tests/arp/test_neighbor_mac_noptf.py:40]: how about change the name as it's returning the count rather then checking.

**Comment by smaheshm** [comment]
looks good.. just minor comments.

File: tests/arp/test_neighbor_mac_noptf.py (hunk 1 above)
Comment (arlakshm) [reply tests/arp/test_neighbor_mac_noptf.py]: fixed

File: tests/arp/test_neighbor_mac_noptf.py (hunk 2 above)
Comment (arlakshm) [reply tests/arp/test_neighbor_mac_noptf.py]: fixed in latest commit

**Review by smaheshm — APPROVED** [review]
:shipit:

**Pull request merged**